### PR TITLE
Pipe mode

### PR DESCRIPTION
**Pull request by nsorense4**
## Merge checklist
- [ ] Docker image or CCS Project builds with the changes
- [ ] Docker image or CCS Project runs with the changes
- [ ] Can send commands from a mocked ground station/relevant changes tested
- [ ] Changes have been reviewed

### CODE REVIEW

File: Services/source/housekeeping/housekeeping_service.c [hunk 1]
```diff
@@ -494,6 +524,7 @@ uint16_t get_size_of_housekeeping(All_systems_housekeeping* all_hk_data) {
 Result write_hk_to_file(uint16_t filenumber, All_systems_housekeeping* all_hk_data) {
   int32_t fout = red_open(fileName, RED_O_CREAT | RED_O_RDWR); //open or create file to write binary
   if (fout == -1) {
+    printf("Unexpected error %d from red_open()\r\n", (int)red_errno);
```
Comment (arrooney) [inline Services/source/housekeeping/housekeeping_service.c:527]: use `ex2_log`

File: Services/include/housekeeping/housekeeping_service.h [hunk 2]
```diff
@@ -31,6 +31,7 @@
 #include "sband.h"
 #include "hyperion.h"
 #include "adcs.h"
+//#include "ex2_hal/ex2_payload_software/hardware_interface/include/payload.h"
```
Comment (arrooney) [inline Services/include/housekeeping/housekeeping_service.h:34]: take this out

File: Services/source/housekeeping/housekeeping_service.c [hunk 3]
```diff
@@ -227,9 +228,16 @@ Result mock_everyone(All_systems_housekeeping* all_hk_data) {
 
   //Athena
   uint8_t i;
-  for (i = 0; i < 6; i++) {
+  for (i = 0; i < 2; i++) {
```
Comment (arrooney) [inline Services/source/housekeeping/housekeeping_service.c:231]: We ought to have a definition of the number of elements in the temperature array.

File: Services/source/housekeeping/housekeeping_service.c [hunk 4]
```diff
@@ -367,13 +375,35 @@ Result mock_everyone(All_systems_housekeeping* all_hk_data) {
...
   
+  #ifndef EPS_IS_STUBBED
+    eps_refresh_instantaneous_telemetry();
+    eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
+
+    //Adding these in order to get the startup telemetry - should be added elsewhere
+    eps_refresh_startup_telemetry();
+    eps_startup_telemetry_t eps_startup = get_eps_startup_telemetry();
+    EPS_getHK(&all_hk_data->EPS_hk,&all_hk_data->EPS_startup_hk);                                   //EPS get housekeeping
+  #endif /* EPS Housekeeping */
```
Comment (arrooney) [inline Services/source/housekeeping/housekeeping_service.c:394]: This comment should state the condition that it ends. `#endif // EPS_IS_STUBBED`

Comment (arrooney) [reply Services/source/housekeeping/housekeeping_service.c:394]: same with the ensuing instances

File: Services/source/housekeeping/housekeeping_service.c [hunk 5]
```diff
@@ -225,13 +235,20 @@ Result mock_everyone(All_systems_housekeeping* all_hk_data) {
   all_hk_data->adcs_hk.Rate_Sensor_Temp_Y = tempFloat;
   all_hk_data->adcs_hk.Rate_Sensor_Temp_Z = tempFloat;
 
-  //Athena
+  // Athena
   uint8_t i;
-  for (i = 0; i < 6; i++) {
+  for (i = 0; i < ATHENA_TEMP_ARRAY_SIZE; i++) {
```
Comment (nsorense4) [inline Services/source/housekeeping/housekeeping_service.c:240]: added this macro. Clearly the clang-format didn't work terribly well. I'll try another commit tomorrow.

File: Services/source/housekeeping/housekeeping_service.c (hunk 1 above)
Comment (nsorense4) [reply Services/source/housekeeping/housekeeping_service.c]: This has been replaced.

File: Services/include/housekeeping/housekeeping_service.h (hunk 2 above)
Comment (nsorense4) [reply Services/include/housekeeping/housekeeping_service.h]: done

File: Services/source/housekeeping/housekeeping_service.c (hunk 4 above)
Comment (nsorense4) [reply Services/source/housekeeping/housekeeping_service.c]: done

File: Services/source/housekeeping/housekeeping_service.c (hunk 5 above)
Comment (nsorense4) [reply Services/source/housekeeping/housekeeping_service.c]: Done

File: Services/source/housekeeping/housekeeping_service.c (hunk 3 above)
Comment (nsorense4) [reply Services/source/housekeeping/housekeeping_service.c]: Done.

**Comment by Taylorr82** [comment]
This branch is pretty far behind master, I recommend a rebase before merging. To make sure things are clear

**Comment by tganley** [comment]
All of the changes from this PR are captured in the prototype_flatsat_rebase branch, which currently has an open PR that includes many more changes. As such, I'm going to close this and we should archive this branch.

**Pull request closed without merging**